docs/spec.md

```diff
@@ -6268,6 +6274,19 @@ If `x` is a value or placeholder, this function is a shortcut for
 `member_name(type(x))`.  If `x` is not a type that has an appropriate member, or
 a value or a placeholder of such a type, returns `None`.
 
+* `same_elementtype_ignoring_bitwidth(x: Type, y: Type) -> bool` checks for the
```

Comment:
Let's align the name of this function with the name of the function that does the conversion (and define semantics in terms of that). E.g. if the latter ends up being called `convert_or_quantize`, then this function could be called `is_convertible_or_quantizable`.

Reply:
Thanks @burmako  for the name suggestion!
Also let me capture your comnent @https://github.com/openxla/stablehlo/pull/1796#discussion_r1353673707 here. 

>> What happens if only one of the types is quantized?

I agree that generalizing `convert_or_quantize` to capture other scenarios will help make this function sufficient  on its own and can be reused in other places, if needed.  

>> Let's align the name of this function with the name of the function that does the conversion (and define semantics in terms of that). E.g. if the latter ends up being called convert_or_quantize, then this function could be called is_convertible_or_quantizable.

Per the [RFC](https://github.com/openxla/stablehlo/pull/1664), the criteria for conversion  is limited to `integer -> integer`, `float -> float`, `complex -> complex` and `quantized_type -> quantized_type`. With that `is_convertible_or_quantizable` might not fit as the pre-requisite for the genetic version of `convert_or_quantize`.
 
One option, which I prefer,  is to use generic `convert_or_quantize` and keep the suggested name "is_convertable_or_quantizable" with limited checking criteria and add rationale to state its limited functionality. 

A few other options considered: 
1. Sync the routines corresponding to "check for conversion" and "conversion"
    - (a) Keep both of them limited to specific conversions only. In that case, change the routine names to something like `convert_for_reduction_computations` and `is_convertible_for_reduction_computation`. However, introducing operation specific functions defeats the generality of such meta functions to be re-usable. 
    - (b) Keep both generalized. Per the RFC, this will include additional scenarios of type conversions which we do not have the use-case for.

2. Keep  "check for conversion" routine as special purpose and the "conversion" routine as generic. 
   - With that we can use `convert_or_quantize` as a generic conversion routine and use some special purpose name like  `is_convertible_for_reduction_computation` for "check for conversion" routine. The difference with 1(a): is that it using one (vs two) operations specific function. 

Please let me know your thoughts?

Reply:
Update:
The current PR is structured around (2),
Suggested name for "check for conversion" routine: `is_promotable` and while defining this function the spec text provided rationale for its specific behavior.  
Suggested name for "conversion" routine: `convert__or_quantize_or_dequantize`.